lib/ExpensiMark.js

```diff
@@ -261,7 +264,9 @@ export default class ExpensiMark {
                 name: 'quote',
                 regex: /\n?<(blockquote|q)(?:"[^"]*"|'[^']*'|[^'">])*>([\s\S]*?)<\/\1>(?![^<]*(<\/pre>|<\/code>))/gi,
                 replacement: (match, g1, g2) => {
-                    const resultString = g2.trim().split('\n').map(m => `> ${m}`).join('\n');
+                    const resultString = g2.replace(/(\[block\])+/g, '\n').trim()
+                        .split('\n').map(m => `> ${m}`)
+                        .join('\n');
```

Comment:
@eh2077 

Why we are doing this is still unclear.

> Currently, we handle line break between markdown block element in method [replaceBlockWithNewLine](https://github.com/Expensify/expensify-common/blob/3cdaa947fe77016206c15e523017cd50678f2359/lib/ExpensiMark.js#L475-L501)

Is there any problem with that? 

Why do we have to do the following?

> While inside a quote, I think it'll be better to handle the [block] separators which wrap the heading here in the replacement method of html-blockquote-to-markdown-quote rule because|
> 
> 1. We'll only have one kind of markdown block element, heading, inside a quote
> 2. It's not that easy to modify method [replaceBlockWithNewLine](https://github.com/Expensify/expensify-common/blob/3cdaa947fe77016206c15e523017cd50678f2359/lib/ExpensiMark.js#L475-L501) to handle heading inside a quote

I'm not following what's are the case and what's the another case you are mentioning?

> When writing the above the comment, I found another case of having two continuous headers, like
> ```
> ===== normal heading =====
> # heading A
> # heading B
> ===== heading in quote =====
> > # heading A inside quote
> > # heading B inside quote
> ```

Reply:
@Santhosh-Sellavel 

> @eh2077
> 
> Why we are doing this is still unclear.
> 
> > Currently, we handle line break between markdown block element in method [replaceBlockWithNewLine](https://github.com/Expensify/expensify-common/blob/3cdaa947fe77016206c15e523017cd50678f2359/lib/ExpensiMark.js#L475-L501)
> 
> Is there any problem with that?

Yes, the method [replaceBlockWithNewLine](https://github.com/Expensify/expensify-common/blob/3cdaa947fe77016206c15e523017cd50678f2359/lib/ExpensiMark.js#L475-L501) will have a problem to handle line break for header inside a quote. Let's use html 
```html
<blockquote><h1>heading</h1>test</blockquote>
```
as an example. The html above is translated into 
```
<blockquote>[block]# heading[block]test</blockquote>
```
by [heading1](https://github.com/Expensify/expensify-common/blob/3cdaa947fe77016206c15e523017cd50678f2359/lib/ExpensiMark.js#L233-L237) rule.
Then the current implementation of [quote](https://github.com/Expensify/expensify-common/blob/3cdaa947fe77016206c15e523017cd50678f2359/lib/ExpensiMark.js#L264) rule translates the blockquote into 
```
> [block]# heading[block]test
```
Last, the above output is translated into 
```
> # heading
test
```
by method [replaceBlockWithNewLine](https://github.com/Expensify/expensify-common/blob/3cdaa947fe77016206c15e523017cd50678f2359/lib/ExpensiMark.js#L475-L501).
The first `[block]` is eliminated occasionally because of quote character `>`, see [this line](https://github.com/Expensify/expensify-common/blob/3cdaa947fe77016206c15e523017cd50678f2359/lib/ExpensiMark.js#L493). The second `[block]` is translated into a line break. See below [code snippet](https://github.com/Expensify/expensify-common/blob/3cdaa947fe77016206c15e523017cd50678f2359/lib/ExpensiMark.js#L493-L497) from method replaceBlockWithNewLine
```js
            // Insert '\n' unless it ends with '\n' or '>' or it's the last [block].
            if (text.match(/[\n|>][>]?[\s]?$/) || index === splitText.length - 1) {
                joinedText += text;
            } else {
                joinedText += `${text}\n`;
            }
```
So we need to fix the replacement of `[block]` for header inside a quote.

> 
> Why do we have to do the following?
> 
> > While inside a quote, I think it'll be better to handle the [block] separators which wrap the heading here in the replacement method of html-blockquote-to-markdown-quote rule because|
> > 
> > 1. We'll only have one kind of markdown block element, heading, inside a quote
> > 2. It's not that easy to modify method [replaceBlockWithNewLine](https://github.com/Expensify/expensify-common/blob/3cdaa947fe77016206c15e523017cd50678f2359/lib/ExpensiMark.js#L475-L501) to handle heading inside a quote

Based on the above explanation, we know that the method replaceBlockWithNewLine relies on character `>` to handle line break, which is possibly conflicted with the syntax of markdown quote. So, my idea here is to try to keep the existing logic of replaceBlockWithNewLine not changed and handle the new use case inside the replacement function of quote. 

> 
> I'm not following what's are the case and what's the another case you are mentioning?
> 
> > When writing the above the comment, I found another case of having two continuous headers, like
> > ```
> > ===== normal heading =====
> > # heading A
> > # heading B
> > ===== heading in quote =====
> > > # heading A inside quote
> > > # heading B inside quote
> > ```

Sorry, my bad I should use a more concrete example. Let's say we have two headers in a quote, like
```
> # heading A
> # heading B
```
Then it'll be translated into html
```html
<blockquote><h1>heading A</h1><h1>heading B</h1></blockquote>
```
When we edit the comment, the headers html is translated into markdown, like
```
> [block]# heading A[block][block]# heading B[block]
```
So, we need to replace `[block][block]` with one line break instead of two using following replacement
```
const resultString = g2.replace(/(\[block\])+/g, '\n').trim()
```